tests/suites/os/tests/secureboot/index.js

```diff
@@ -1,145 +1,197 @@
 'use strict';
 
-module.exports = {
-	title: 'secure boot tests',
-	tests: [
-		{
-			title: 'check secure boot',
-			run: async function(test) {
-				const securebootEfiVarPath = '/sys/firmware/efi/efivars/SecureBoot-*';
+const securebootEfiVarPath = '/sys/firmware/efi/efivars/SecureBoot-*';
+const Bluebird = require('bluebird');
 
-				const securebootSupported = await this.worker.executeCommandInHostOS(
-					['ls', securebootEfiVarPath, '&>/dev/null',
-						'&&', 'echo', 'true',
-						'||', 'echo', 'false'
-					],
-					this.link,
-				).then(output => { return output === 'true'});
-
-				// https://wiki.archlinux.org/title/Unified_Extensible_Firmware_Interface/Secure_Boot#Checking_Secure_Boot_status
-				const securebootEnabled = await this.worker.executeCommandInHostOS(
-						[
-							'od', '--address-radix=n', '--format=u1', securebootEfiVarPath,
-								'|', 'tr', '-s', "' '",
-								'|', 'cut', "-d' '", '-f6'
-						],
-						this.link,
-				).then(output => output === '1' );
-
-				// Used after tampering with verified boot files
-				const resetWorker = async () => {
-					await this.worker.off();
-					await this.worker.flash(this.os.image.path);
-					await this.worker.on();
-
-					await this.worker.addSSHKey(this.sshKeyPath);
-				}
-
-				const waitForSerialOutput = async(pattern, slice=0, retries=60, delay=1000) => this.utils.waitUntil(
-					async () => this.worker.fetchSerial().then(
-						serialLogs => serialLogs.split('\n')
-																		.slice(slice)
-																		.join('\n')
-																		.match(pattern)
-					), false, retries, delay
-				);
-
-				const testFullDiskEncryption = async () => Promise.all(
-					[
-						{ pattern: '/^\\/mnt\\/boot$/', name: 'Boot partition' },
-						{ pattern: '/^\\/mnt\\/sysroot\\/active$/', name: 'Root partition' },
-						{ pattern: '/^\\/mnt\\/state$/', name: 'State partition' },
-						{ pattern: '/^\\/mnt\\/data$/', name: 'Data partition' },
-					].map(args => test.resolveMatch(
-							this.worker.executeCommandInHostOS(
-								[
-									'lsblk', '-nlo', 'MOUNTPOINT,TYPE',
-									'|', 'awk', `'$1 ~ ${args.pattern} { print $2 }'`
-								],
-								this.link,
-							),
-							/crypt/,
-							`${args.name} is encrypted`,
-						)
-					)
-				);
+class secureBoot {
+	constructor(test, worker, context, imagePath) {
+		this.test = test;
+		this.worker = worker;
+		this.link = context.get().link;
+		this.imagePath = imagePath;
+		this.sshKeyPath = context.get().sshKeyPath;
+		this.utils = context.get().utils;
+	}
 
-				const testModuleVerification = async() => {
-					await test.resolves(
-						this.worker.executeCommandInHostOS('modprobe pcan_netdev', this.link),
-						'Module with valid signature loads',
-					);
+	async resetWorker() {
+		await this.worker.off();
+		await this.worker.flash(this.imagePath);
```

Comment:
One thing I'm still unsure about when we have to test on other types of device is to what extend we will need to manipulate the flashing process - do you forsee that we might need to pass other parameters here, other than the OS image? (this is an issue separate but related to this PR)

Reply:
Unless the `flash` method has a way to determine what device it is running on to decide how to program we might need to specialize the `resetWorker()/resetDUT()` function.